src/main/java/org/candlepin/subscriptions/tally/facts/FactNormalizer.java

```diff
@@ -76,9 +77,24 @@ public NormalizedFacts normalize(InventoryHostFacts hostFacts) {
         normalizeQpcFacts(normalizedFacts, hostFacts);
         normalizeSocketCount(normalizedFacts);
         normalizeConflictingOrMissingRhelVariants(normalizedFacts);
+        pruneProducts(normalizedFacts);
         return normalizedFacts;
     }
 
+    @SuppressWarnings("indentation")
+    private void pruneProducts(NormalizedFacts normalizedFacts) {
+        // If a Satellite product was found, do not include RHEL or its variants.
+        boolean hasSatellite = normalizedFacts.getProducts().stream()
+            .anyMatch(s -> s.matches("^.*[Ss]atellite.*$"));
```

Comment:
Since we control the names of the extracted products, I don't think we need to be quite so flexible. Specifically, I'd like to suggest that we simply go ahead and pick names for the Satellite variants. In #64 I suggested `Satellite 6` and `Satellite 6 Capsule`. That'd eliminate the need for regex, and you could just go ahead and use `s.startsWith('Satellite')`

Reply:
DONE